--- lua-5.3.5.orig/src/lua.c
+++ lua-5.3.5/src/lua.c
@@ -570,6 +570,8 @@
   }
   luaL_openlibs(L);  /* open standard libraries */
   createargtable(L, argv, argc, script);  /* create table 'arg' */
+  if (luaL_dofile(L, LUA_LDIR "init.lua") != LUA_OK)
+    return 0;
   if (!(args & has_E)) {  /* no option '-E'? */
     if (handle_luainit(L) != LUA_OK)  /* run LUA_INIT */
       return 0;  /* error running LUA_INIT */
