--- lua-5.3.5.orig/src/luaconf.h
+++ lua-5.3.5/src/luaconf.h
@@ -200,15 +200,14 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
-#define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
-#define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
+#define LUA_ROOT	"/"
+#define LUA_LDIR	LUA_ROOT "lib/lua/"
+#define LUA_CDIR	""
 #define LUA_PATH_DEFAULT  \
 		LUA_LDIR"?.lua;"  LUA_LDIR"?/init.lua;" \
 		LUA_CDIR"?.lua;"  LUA_CDIR"?/init.lua;" \
 		"./?.lua;" "./?/init.lua"
-#define LUA_CPATH_DEFAULT \
-		LUA_CDIR"?.so;" LUA_CDIR"loadall.so;" "./?.so"
+#define LUA_CPATH_DEFAULT ""
 #endif			/* } */
 
 
