Index: ubase/ubase/getty.c
===================================================================
--- ubase.orig/ubase/getty.c	2023-06-18 11:45:55.993142824 +0200
+++ ubase/ubase/getty.c	2023-06-18 11:47:50.456473330 +0200
@@ -15,8 +15,8 @@
 #include "config.h"
 #include "util.h"
 
-static char *tty = "/dev/tty1";
-static char *defaultterm = "linux";
+static char *tty = "/dev/uart";
+static char *defaultterm = "xterm";
 
 static void
 usage(void)
@@ -28,7 +28,6 @@
 main(int argc, char *argv[])
 {
 	char term[128], logname[LOGIN_NAME_MAX], c;
-	char hostname[HOST_NAME_MAX + 1];
 	struct utmp usr;
 	struct sigaction sa;
 	FILE *fp;
@@ -58,7 +57,7 @@
 
 	setsid();
 
-	fd = open(tty, O_RDWR);
+	fd = open(tty, O_RDWR | O_NOCTTY);
 	if (fd < 0)
 		eprintf("open %s:", tty);
 	if (isatty(fd) == 0)
@@ -67,19 +66,10 @@
 	/* steal the controlling terminal if necessary */
 	if (ioctl(fd, TIOCSCTTY, (void *)1) != 0)
 		weprintf("TIOCSCTTY: could not set controlling tty\n");
-	vhangup();
-	close(fd);
 
-	fd = open(tty, O_RDWR);
-	if (fd < 0)
-		eprintf("open %s:", tty);
 	dup2(fd, 0);
 	dup2(fd, 1);
 	dup2(fd, 2);
-	if (fchown(fd, 0, 0) < 0)
-		weprintf("fchown %s:", tty);
-	if (fchmod(fd, 0600) < 0)
-		weprintf("fchmod %s:", tty);
 	if (fd > 2)
 		close(fd);
 
@@ -112,13 +102,11 @@
 	if (argc > 2)
 		return execvp(argv[2], argv + 2);
 
-	if (gethostname(hostname, sizeof(hostname)) == 0)
-		printf("%s ", hostname);
 	printf("login: ");
 	fflush(stdout);
 
 	/* Flush pending input */
-	ioctl(0, TCFLSH, (void *)0);
+	ioctl(0, TIOCFLUSH, (void *)0);
 	memset(logname, 0, sizeof(logname));
 	while (1) {
 		n = read(0, &c, 1);
