Index: ubase/ubase/getty.c
===================================================================
--- ubase.orig/ubase/getty.c
+++ ubase/ubase/getty.c
@@ -10,13 +10,12 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <utmp.h>
 
 #include "config.h"
 #include "util.h"
 
-static char *tty = "/dev/tty1";
-static char *defaultterm = "linux";
+static char *tty = "/dev/uart";
+static char *defaultterm = "xterm";
 
 static void
 usage(void)
@@ -28,14 +27,10 @@ int
 main(int argc, char *argv[])
 {
 	char term[128], logname[LOGIN_NAME_MAX], c;
-	char hostname[HOST_NAME_MAX + 1];
-	struct utmp usr;
 	struct sigaction sa;
-	FILE *fp;
 	int fd;
 	unsigned int i = 0;
 	ssize_t n;
-	long pos;
 
 	ARGBEGIN {
 	default:
@@ -58,7 +53,7 @@ main(int argc, char *argv[])
 
 	setsid();
 
-	fd = open(tty, O_RDWR);
+	fd = open(tty, O_RDWR | O_NOCTTY);
 	if (fd < 0)
 		eprintf("open %s:", tty);
 	if (isatty(fd) == 0)
@@ -67,19 +62,10 @@ main(int argc, char *argv[])
 	/* steal the controlling terminal if necessary */
 	if (ioctl(fd, TIOCSCTTY, (void *)1) != 0)
 		weprintf("TIOCSCTTY: could not set controlling tty\n");
-	vhangup();
-	close(fd);
 
-	fd = open(tty, O_RDWR);
-	if (fd < 0)
-		eprintf("open %s:", tty);
 	dup2(fd, 0);
 	dup2(fd, 1);
 	dup2(fd, 2);
-	if (fchown(fd, 0, 0) < 0)
-		weprintf("fchown %s:", tty);
-	if (fchmod(fd, 0600) < 0)
-		weprintf("fchmod %s:", tty);
 	if (fd > 2)
 		close(fd);
 
@@ -88,37 +74,14 @@ main(int argc, char *argv[])
 	sigemptyset(&sa.sa_mask);
 	sigaction(SIGHUP, &sa, NULL);
 
-	/* Clear all utmp entries for this tty */
-	fp = fopen(UTMP_PATH, "r+");
-	if (fp) {
-		do {
-			pos = ftell(fp);
-			if (fread(&usr, sizeof(usr), 1, fp) != 1)
-				break;
-			if (usr.ut_line[0] == '\0')
-				continue;
-			if (strcmp(usr.ut_line, tty) != 0)
-				continue;
-			memset(&usr, 0, sizeof(usr));
-			fseek(fp, pos, SEEK_SET);
-			if (fwrite(&usr, sizeof(usr), 1, fp) != 1)
-				break;
-		} while (1);
-		if (ferror(fp))
-			weprintf("%s: I/O error:", UTMP_PATH);
-		fclose(fp);
-	}
-
 	if (argc > 2)
 		return execvp(argv[2], argv + 2);
 
-	if (gethostname(hostname, sizeof(hostname)) == 0)
-		printf("%s ", hostname);
 	printf("login: ");
 	fflush(stdout);
 
 	/* Flush pending input */
-	ioctl(0, TCFLSH, (void *)0);
+	ioctl(0, TIOCFLUSH, (void *)0);
 	memset(logname, 0, sizeof(logname));
 	while (1) {
 		n = read(0, &c, 1);
