Index: ubase/ubase/su.c
===================================================================
--- ubase.orig/ubase/su.c	2023-05-30 17:51:24.363395446 +0200
+++ ubase/ubase/su.c	2023-05-30 17:55:05.003401122 +0200
@@ -14,6 +14,7 @@
 #include "util.h"
 
 extern char **environ;
+extern char *envinit[1];
 
 static int lflag = 0;
 static int pflag = 0;
@@ -21,9 +22,9 @@
 static int
 dologin(struct passwd *pw)
 {
-	char *shell = pw->pw_shell[0] == '\0' ? "/bin/sh" : pw->pw_shell;
+	const char *shell = pw->pw_shell[0] == '\0' ? "/bin/sh" : pw->pw_shell;
 	char *term = getenv("TERM");
-	clearenv();
+	environ = envinit;
 	setenv("HOME", pw->pw_dir, 1);
 	setenv("SHELL", shell, 1);
 	setenv("USER", pw->pw_name, 1);
@@ -50,7 +51,7 @@
 main(int argc, char *argv[])
 {
 	char *usr = "root", *pass;
-	char *shell;
+	const char *shell;
 	struct passwd *pw;
 	char *newargv[2];
 	uid_t uid;
@@ -102,7 +103,7 @@
 		return dologin(pw);
 	} else {
 		shell = pw->pw_shell[0] == '\0' ? "/bin/sh" : pw->pw_shell;
-		newargv[0] = shell;
+		newargv[0] = (char *)shell;
 		newargv[1] = NULL;
 		if (!pflag) {
 			setenv("HOME", pw->pw_dir, 1);
