Index: ubase/ubase/util.h
===================================================================
--- ubase.orig/ubase/util.h	2023-06-19 19:59:09.923110619 +0200
+++ ubase/ubase/util.h	2023-06-19 20:14:22.103080495 +0200
@@ -12,6 +12,13 @@
 
 #define LEN(x) (sizeof (x) / sizeof *(x))
 
+extern char** environ;
+extern char* envinit[1];
+
+static inline void clearenv(void) {
+    environ = envinit;
+}
+
 /* eprintf.c */
 extern char *argv0;
 
