Index: lua/lua/lua.c
===================================================================
--- lua.orig/lua/lua.c
+++ lua/lua/lua.c
@@ -623,6 +623,8 @@ static int pmain (lua_State *L) {
   }
   luaL_openlibs(L);  /* open standard libraries */
   createargtable(L, argv, argc, script);  /* create table 'arg' */
+  if (luaL_dofile(L, LUA_LDIR "init.lua") != LUA_OK)
+    return 0;
   lua_gc(L, LUA_GCGEN, 0, 0);  /* GC in generational mode */
   if (!(args & has_E)) {  /* no option '-E'? */
     if (handle_luainit(L) != LUA_OK)  /* run LUA_INIT */
