Index: lua/lua/linit.c
===================================================================
--- lua.orig/lua/linit.c
+++ lua/lua/linit.c
@@ -50,6 +50,7 @@ static const luaL_Reg loadedlibs[] = {
   {LUA_MATHLIBNAME, luaopen_math},
   {LUA_UTF8LIBNAME, luaopen_utf8},
   {LUA_DBLIBNAME, luaopen_debug},
+  {LUA_UNIXNAME, luaopen_unix},
 #if defined(LUA_COMPAT_BITLIB)
   {LUA_BITLIBNAME, luaopen_bit32},
 #endif
Index: lua/lua/lualib.h
===================================================================
--- lua.orig/lua/lualib.h
+++ lua/lua/lualib.h
@@ -44,6 +44,9 @@ LUAMOD_API int (luaopen_math) (lua_State
 #define LUA_DBLIBNAME	"debug"
 LUAMOD_API int (luaopen_debug) (lua_State *L);
 
+#define LUA_UNIXNAME "unix"
+LUAMOD_API int (luaopen_unix) (lua_State *L);
+
 #define LUA_LOADLIBNAME	"package"
 LUAMOD_API int (luaopen_package) (lua_State *L);
 
Index: lua/lua/makefile
===================================================================
--- lua.orig/lua/makefile
+++ lua/lua/makefile
@@ -77,7 +77,7 @@ CORE_O=	lapi.o lcode.o lctype.o ldebug.o
 	ltm.o lundump.o lvm.o lzio.o ltests.o
 AUX_O=	lauxlib.o
 LIB_O=	lbaselib.o ldblib.o liolib.o lmathlib.o loslib.o ltablib.o lstrlib.o \
-	lutf8lib.o lbitlib.o loadlib.o lcorolib.o linit.o
+	lutf8lib.o lbitlib.o loadlib.o lcorolib.o linit.o lunix.o
 
 LUA_T=	lua
 LUA_O=	lua.o
