Index: sbase/sbase/basename.c
===================================================================
--- sbase.orig/sbase/basename.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/basename.c	2022-12-06 20:18:50.939624592 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+basename_main(int argc, char *argv[])
 {
 	ssize_t off;
 	char *p;
Index: sbase/sbase/cal.c
===================================================================
--- sbase.orig/sbase/cal.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/cal.c	2022-12-06 20:18:50.946291258 +0100
@@ -146,7 +146,7 @@
 }
 
 int
-main(int argc, char *argv[])
+cal_main(int argc, char *argv[])
 {
 	time_t now;
 	size_t year, ncols, nmons;
Index: sbase/sbase/cat.c
===================================================================
--- sbase.orig/sbase/cat.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/cat.c	2022-12-06 20:18:50.949624592 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+cat_main(int argc, char *argv[])
 {
 	int fd, ret = 0;
 
Index: sbase/sbase/chgrp.c
===================================================================
--- sbase.orig/sbase/chgrp.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/chgrp.c	2022-12-06 20:18:50.952957925 +0100
@@ -35,7 +35,7 @@
 }
 
 int
-main(int argc, char *argv[])
+chgrp_main(int argc, char *argv[])
 {
 	struct group *gr;
 	struct recursor r = { .fn = chgrp, .maxdepth = 1, .follow = 'P' };
Index: sbase/sbase/chmod.c
===================================================================
--- sbase.orig/sbase/chmod.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/chmod.c	2022-12-06 20:18:50.956291258 +0100
@@ -30,7 +30,7 @@
 }
 
 int
-main(int argc, char *argv[])
+chmod_main(int argc, char *argv[])
 {
 	struct recursor r = { .fn = chmodr, .maxdepth = 1, .follow = 'H', .flags = DIRFIRST };
 	size_t i;
Index: sbase/sbase/chown.c
===================================================================
--- sbase.orig/sbase/chown.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/chown.c	2022-12-06 20:18:50.959624591 +0100
@@ -41,7 +41,7 @@
 }
 
 int
-main(int argc, char *argv[])
+chown_main(int argc, char *argv[])
 {
 	struct group *gr;
 	struct passwd *pw;
Index: sbase/sbase/chroot.c
===================================================================
--- sbase.orig/sbase/chroot.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/chroot.c	2022-12-06 20:18:50.962957924 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+chroot_main(int argc, char *argv[])
 {
 	char *shell[] = { "/bin/sh", "-i", NULL }, *aux, *cmd;
 	int savederrno;
Index: sbase/sbase/cksum.c
===================================================================
--- sbase.orig/sbase/cksum.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/cksum.c	2022-12-06 20:18:50.962957924 +0100
@@ -99,7 +99,7 @@
 }
 
 int
-main(int argc, char *argv[])
+cksum_main(int argc, char *argv[])
 {
 	int fd;
 
Index: sbase/sbase/cmp.c
===================================================================
--- sbase.orig/sbase/cmp.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/cmp.c	2022-12-06 20:18:50.966291258 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+cmp_main(int argc, char *argv[])
 {
 	FILE *fp[2];
 	size_t line = 1, n;
Index: sbase/sbase/cols.c
===================================================================
--- sbase.orig/sbase/cols.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/cols.c	2022-12-06 20:18:50.969624591 +0100
@@ -18,7 +18,7 @@
 }
 
 int
-main(int argc, char *argv[])
+cols_main(int argc, char *argv[])
 {
 	FILE *fp;
 	struct winsize w;
Index: sbase/sbase/comm.c
===================================================================
--- sbase.orig/sbase/comm.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/comm.c	2022-12-06 20:18:50.976291258 +0100
@@ -30,7 +30,7 @@
 }
 
 int
-main(int argc, char *argv[])
+comm_main(int argc, char *argv[])
 {
 	FILE *fp[2];
 	static struct line line[2];
Index: sbase/sbase/cp.c
===================================================================
--- sbase.orig/sbase/cp.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/cp.c	2022-12-06 20:18:50.979624590 +0100
@@ -11,7 +11,7 @@
 }
 
 int
-main(int argc, char *argv[])
+cp_main(int argc, char *argv[])
 {
 	struct stat st;
 
Index: sbase/sbase/cron.c
===================================================================
--- sbase.orig/sbase/cron.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/cron.c	2022-12-06 20:18:50.982957924 +0100
@@ -484,7 +484,7 @@
 }
 
 int
-main(int argc, char *argv[])
+cron_main(int argc, char *argv[])
 {
 	FILE *fp;
 	struct ctabentry *cte;
Index: sbase/sbase/cut.c
===================================================================
--- sbase.orig/sbase/cut.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/cut.c	2022-12-06 20:18:50.986291257 +0100
@@ -160,7 +160,7 @@
 }
 
 int
-main(int argc, char *argv[])
+cut_main(int argc, char *argv[])
 {
 	FILE *fp;
 	int ret = 0;
Index: sbase/sbase/date.c
===================================================================
--- sbase.orig/sbase/date.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/date.c	2022-12-06 20:18:50.989624590 +0100
@@ -62,7 +62,7 @@
 }
 
 int
-main(int argc, char *argv[])
+date_main(int argc, char *argv[])
 {
 	struct tm *now;
 	time_t t;
Index: sbase/sbase/dd.c
===================================================================
--- sbase.orig/sbase/dd.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/dd.c	2022-12-06 20:18:50.992957924 +0100
@@ -72,7 +72,7 @@
 }
 
 int
-main(int argc, char *argv[])
+dd_main(int argc, char *argv[])
 {
 	enum {
 		LCASE   = 1 << 0,
Index: sbase/sbase/dirname.c
===================================================================
--- sbase.orig/sbase/dirname.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/dirname.c	2022-12-06 20:18:50.996291257 +0100
@@ -11,7 +11,7 @@
 }
 
 int
-main(int argc, char *argv[])
+dirname_main(int argc, char *argv[])
 {
 	ARGBEGIN {
 	default:
Index: sbase/sbase/du.c
===================================================================
--- sbase.orig/sbase/du.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/du.c	2022-12-06 20:18:50.999624590 +0100
@@ -58,7 +58,7 @@
 }
 
 int
-main(int argc, char *argv[])
+du_main(int argc, char *argv[])
 {
 	struct recursor r = { .fn = du, .follow = 'P' };
 	off_t n = 0;
Index: sbase/sbase/echo.c
===================================================================
--- sbase.orig/sbase/echo.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/echo.c	2022-12-06 20:18:51.002957924 +0100
@@ -4,7 +4,7 @@
 #include "util.h"
 
 int
-main(int argc, char *argv[])
+echo_main(int argc, char *argv[])
 {
 	int nflag = 0;
 
Index: sbase/sbase/ed.c
===================================================================
--- sbase.orig/sbase/ed.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/ed.c	2022-12-06 20:18:51.002957924 +0100
@@ -1442,7 +1442,7 @@
 }
 
 int
-main(int argc, char *argv[])
+ed_main(int argc, char *argv[])
 {
 	ARGBEGIN {
 	case 'p':
Index: sbase/sbase/env.c
===================================================================
--- sbase.orig/sbase/env.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/env.c	2022-12-06 20:18:51.006291256 +0100
@@ -16,7 +16,7 @@
 }
 
 int
-main(int argc, char *argv[])
+env_main(int argc, char *argv[])
 {
 	int savederrno;
 
Index: sbase/sbase/expand.c
===================================================================
--- sbase.orig/sbase/expand.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/expand.c	2022-12-06 20:18:51.009624590 +0100
@@ -86,7 +86,7 @@
 }
 
 int
-main(int argc, char *argv[])
+expand_main(int argc, char *argv[])
 {
 	FILE *fp;
 	int ret = 0;
Index: sbase/sbase/expr.c
===================================================================
--- sbase.orig/sbase/expr.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/expr.c	2022-12-06 20:18:51.012957923 +0100
@@ -255,7 +255,7 @@
 }
 
 int
-main(int argc, char *argv[])
+expr_main(int argc, char *argv[])
 {
 	int ret;
 
Index: sbase/sbase/false.c
===================================================================
--- sbase.orig/sbase/false.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/false.c	2022-12-06 20:18:51.016291256 +0100
@@ -1,6 +1,6 @@
 /* See LICENSE file for copyright and license details. */
 int
-main(void)
+false_main(void)
 {
 	return 1;
 }
Index: sbase/sbase/find.c
===================================================================
--- sbase.orig/sbase/find.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/find.c	2022-12-06 20:18:51.016291256 +0100
@@ -1011,7 +1011,7 @@
 }
 
 int
-main(int argc, char **argv)
+find_main(int argc, char **argv)
 {
 	char **paths;
 	int npaths;
Index: sbase/sbase/flock.c
===================================================================
--- sbase.orig/sbase/flock.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/flock.c	2022-12-06 20:18:51.019624590 +0100
@@ -16,7 +16,7 @@
 }
 
 int
-main(int argc, char *argv[])
+flock_main(int argc, char *argv[])
 {
 	int fd, status, savederrno, flags = LOCK_EX, nonblk = 0, oflag = 0;
 	pid_t pid;
Index: sbase/sbase/fold.c
===================================================================
--- sbase.orig/sbase/fold.c	2022-12-06 19:10:45.119759804 +0100
+++ sbase/sbase/fold.c	2022-12-06 20:18:51.022957922 +0100
@@ -83,7 +83,7 @@
 }
 
 int
-main(int argc, char *argv[])
+fold_main(int argc, char *argv[])
 {
 	FILE *fp;
 	int ret = 0;
Index: sbase/sbase/getconf.c
===================================================================
--- sbase.orig/sbase/getconf.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/getconf.c	2022-12-06 20:18:51.026291256 +0100
@@ -21,7 +21,7 @@
 }
 
 int
-main(int argc, char *argv[])
+getconf_main(int argc, char *argv[])
 {
 	size_t len;
 	long res;
Index: sbase/sbase/grep.c
===================================================================
--- sbase.orig/sbase/grep.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/grep.c	2022-12-06 20:18:51.029624589 +0100
@@ -173,7 +173,7 @@
 }
 
 int
-main(int argc, char *argv[])
+grep_main(int argc, char *argv[])
 {
 	struct pattern *pnode;
 	int m, flags = REG_NOSUB, match = NoMatch;
Index: sbase/sbase/head.c
===================================================================
--- sbase.orig/sbase/head.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/head.c	2022-12-06 20:18:51.029624589 +0100
@@ -29,7 +29,7 @@
 }
 
 int
-main(int argc, char *argv[])
+head_main(int argc, char *argv[])
 {
 	size_t n = 10;
 	FILE *fp;
Index: sbase/sbase/hostname.c
===================================================================
--- sbase.orig/sbase/hostname.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/hostname.c	2022-12-06 20:18:51.032957922 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+hostname_main(int argc, char *argv[])
 {
 	char host[HOST_NAME_MAX + 1];
 
Index: sbase/sbase/join.c
===================================================================
--- sbase.orig/sbase/join.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/join.c	2022-12-06 20:18:51.036291256 +0100
@@ -450,7 +450,7 @@
 
 
 int
-main(int argc, char *argv[])
+join_main(int argc, char *argv[])
 {
 	size_t jf[2] = { jfield, jfield, };
 	FILE *fp[2];
Index: sbase/sbase/kill.c
===================================================================
--- sbase.orig/sbase/kill.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/kill.c	2022-12-06 20:18:51.039624588 +0100
@@ -56,7 +56,7 @@
 }
 
 int
-main(int argc, char *argv[])
+kill_main(int argc, char *argv[])
 {
 	pid_t pid;
 	size_t i;
Index: sbase/sbase/link.c
===================================================================
--- sbase.orig/sbase/link.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/link.c	2022-12-06 20:18:51.042957922 +0100
@@ -10,7 +10,7 @@
 }
 
 int
-main(int argc, char *argv[])
+link_main(int argc, char *argv[])
 {
 	ARGBEGIN {
 	default:
Index: sbase/sbase/ln.c
===================================================================
--- sbase.orig/sbase/ln.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/ln.c	2022-12-06 20:18:51.042957922 +0100
@@ -17,7 +17,7 @@
 }
 
 int
-main(int argc, char *argv[])
+ln_main(int argc, char *argv[])
 {
 	char *targetdir = ".", *target = NULL;
 	int ret = 0, sflag = 0, fflag = 0, dirfd = AT_FDCWD,
Index: sbase/sbase/logger.c
===================================================================
--- sbase.orig/sbase/logger.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/logger.c	2022-12-06 20:18:51.046291256 +0100
@@ -44,7 +44,7 @@
 }
 
 int
-main(int argc, char *argv[])
+logger_main(int argc, char *argv[])
 {
 	size_t sz;
 	int logflags = 0, priority = LOG_NOTICE, i;
Index: sbase/sbase/logname.c
===================================================================
--- sbase.orig/sbase/logname.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/logname.c	2022-12-06 20:18:51.049624588 +0100
@@ -11,7 +11,7 @@
 }
 
 int
-main(int argc, char *argv[])
+logname_main(int argc, char *argv[])
 {
 	char *login;
 
Index: sbase/sbase/ls.c
===================================================================
--- sbase.orig/sbase/ls.c	2022-12-06 19:10:45.123093136 +0100
+++ sbase/sbase/ls.c	2022-12-06 20:18:51.052957922 +0100
@@ -366,7 +366,7 @@
 }
 
 int
-main(int argc, char *argv[])
+ls_main(int argc, char *argv[])
 {
 	struct entry ent, *dents, *fents;
 	size_t i, ds, fs;
Index: sbase/sbase/md5sum.c
===================================================================
--- sbase.orig/sbase/md5sum.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/md5sum.c	2022-12-06 20:18:51.056291255 +0100
@@ -22,7 +22,7 @@
 }
 
 int
-main(int argc, char *argv[])
+md5sum_main(int argc, char *argv[])
 {
 	int ret = 0, (*cryptfunc)(int, char **, struct crypt_ops *, uint8_t *, size_t) = cryptmain;
 	uint8_t md[MD5_DIGEST_LENGTH];
Index: sbase/sbase/mkdir.c
===================================================================
--- sbase.orig/sbase/mkdir.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/mkdir.c	2022-12-06 20:18:51.056291255 +0100
@@ -13,7 +13,7 @@
 }
 
 int
-main(int argc, char *argv[])
+mkdir_main(int argc, char *argv[])
 {
 	mode_t mode, mask;
 	int pflag = 0, ret = 0;
Index: sbase/sbase/mkfifo.c
===================================================================
--- sbase.orig/sbase/mkfifo.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/mkfifo.c	2022-12-06 20:18:51.059624588 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+mkfifo_main(int argc, char *argv[])
 {
 	mode_t mode = 0666;
 	int ret = 0;
Index: sbase/sbase/mknod.c
===================================================================
--- sbase.orig/sbase/mknod.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/mknod.c	2022-12-06 20:18:51.062957922 +0100
@@ -22,7 +22,7 @@
 }
 
 int
-main(int argc, char *argv[])
+mknod_main(int argc, char *argv[])
 {
 	mode_t mode = 0666;
 	dev_t dev;
Index: sbase/sbase/mktemp.c
===================================================================
--- sbase.orig/sbase/mktemp.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/mktemp.c	2022-12-06 20:18:51.066291254 +0100
@@ -14,7 +14,7 @@
 }
 
 int
-main(int argc, char *argv[])
+mktemp_main(int argc, char *argv[])
 {
 	int dflag = 0, pflag = 0, qflag = 0, tflag = 0, uflag = 0, fd;
 	char *template = "tmp.XXXXXXXXXX", *tmpdir = "", *pdir,
Index: sbase/sbase/mv.c
===================================================================
--- sbase.orig/sbase/mv.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/mv.c	2022-12-06 20:18:51.069624588 +0100
@@ -42,7 +42,7 @@
 }
 
 int
-main(int argc, char *argv[])
+mv_main(int argc, char *argv[])
 {
 	struct stat st;
 
Index: sbase/sbase/nice.c
===================================================================
--- sbase.orig/sbase/nice.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/nice.c	2022-12-06 20:18:51.069624588 +0100
@@ -22,7 +22,7 @@
 }
 
 int
-main(int argc, char *argv[])
+nice_main(int argc, char *argv[])
 {
 	int val = 10, r, savederrno;
 
Index: sbase/sbase/nl.c
===================================================================
--- sbase.orig/sbase/nl.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/nl.c	2022-12-06 20:18:51.072957921 +0100
@@ -116,7 +116,7 @@
 }
 
 int
-main(int argc, char *argv[])
+nl_main(int argc, char *argv[])
 {
 	FILE *fp = NULL;
 	size_t s;
Index: sbase/sbase/nohup.c
===================================================================
--- sbase.orig/sbase/nohup.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/nohup.c	2022-12-06 20:18:51.076291254 +0100
@@ -15,7 +15,7 @@
 }
 
 int
-main(int argc, char *argv[])
+nohup_main(int argc, char *argv[])
 {
 	int fd, savederrno;
 
Index: sbase/sbase/od.c
===================================================================
--- sbase.orig/sbase/od.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/od.c	2022-12-06 20:18:51.079624588 +0100
@@ -208,7 +208,7 @@
 }
 
 int
-main(int argc, char *argv[])
+od_main(int argc, char *argv[])
 {
 	int fd;
 	struct type *t;
Index: sbase/sbase/paste.c
===================================================================
--- sbase.orig/sbase/paste.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/paste.c	2022-12-06 20:18:51.082957920 +0100
@@ -85,7 +85,7 @@
 }
 
 int
-main(int argc, char *argv[])
+paste_main(int argc, char *argv[])
 {
 	struct fdescr *dsc;
 	Rune *delim_rune = NULL;
Index: sbase/sbase/pathchk.c
===================================================================
--- sbase.orig/sbase/pathchk.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/pathchk.c	2022-12-06 20:18:51.082957920 +0100
@@ -79,7 +79,7 @@
 }
 
 int
-main(int argc, char *argv[])
+pathchk_main(int argc, char *argv[])
 {
 	int ret = 0;
 
Index: sbase/sbase/printenv.c
===================================================================
--- sbase.orig/sbase/printenv.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/printenv.c	2022-12-06 20:18:51.086291254 +0100
@@ -13,7 +13,7 @@
 }
 
 int
-main(int argc, char *argv[])
+printenv_main(int argc, char *argv[])
 {
 	char *var;
 	int ret = 0;
Index: sbase/sbase/printf.c
===================================================================
--- sbase.orig/sbase/printf.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/printf.c	2022-12-06 20:18:51.089624587 +0100
@@ -16,7 +16,7 @@
 }
 
 int
-main(int argc, char *argv[])
+printf_main(int argc, char *argv[])
 {
 	Rune *rarg;
 	size_t i, j, argi, lastargi, formatlen, blen;
Index: sbase/sbase/pwd.c
===================================================================
--- sbase.orig/sbase/pwd.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/pwd.c	2022-12-06 20:18:51.092957920 +0100
@@ -28,7 +28,7 @@
 }
 
 int
-main(int argc, char *argv[])
+pwd_main(int argc, char *argv[])
 {
 	char cwd[PATH_MAX];
 	char mode = 'L';
Index: sbase/sbase/readlink.c
===================================================================
--- sbase.orig/sbase/readlink.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/readlink.c	2022-12-06 20:18:51.096291254 +0100
@@ -14,7 +14,7 @@
 }
 
 int
-main(int argc, char *argv[])
+readlink_main(int argc, char *argv[])
 {
 	char buf[PATH_MAX];
 	ssize_t n;
Index: sbase/sbase/renice.c
===================================================================
--- sbase.orig/sbase/renice.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/renice.c	2022-12-06 20:18:51.096291254 +0100
@@ -41,7 +41,7 @@
 }
 
 int
-main(int argc, char *argv[])
+renice_main(int argc, char *argv[])
 {
 	const char *adj = NULL;
 	long val;
Index: sbase/sbase/rev.c
===================================================================
--- sbase.orig/sbase/rev.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/rev.c	2022-12-06 20:18:51.099624586 +0100
@@ -40,7 +40,7 @@
 }
 
 int
-main(int argc, char *argv[])
+rev_main(int argc, char *argv[])
 {
 	FILE *fp;
 	int ret = 0;
Index: sbase/sbase/rm.c
===================================================================
--- sbase.orig/sbase/rm.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/rm.c	2022-12-06 20:18:51.102957920 +0100
@@ -11,7 +11,7 @@
 }
 
 int
-main(int argc, char *argv[])
+rm_main(int argc, char *argv[])
 {
 	struct recursor r = { .fn = rm, .maxdepth = 1, .follow = 'P' };
 
Index: sbase/sbase/rmdir.c
===================================================================
--- sbase.orig/sbase/rmdir.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/rmdir.c	2022-12-06 20:18:51.106291253 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+rmdir_main(int argc, char *argv[])
 {
 	int pflag = 0, ret = 0;
 	char *d;
Index: sbase/sbase/sed.c
===================================================================
--- sbase.orig/sbase/sed.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sed.c	2022-12-06 20:18:51.109624586 +0100
@@ -1690,7 +1690,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sed_main(int argc, char *argv[])
 {
 	char *arg;
 	int script = 0;
Index: sbase/sbase/seq.c
===================================================================
--- sbase.orig/sbase/seq.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/seq.c	2022-12-06 20:18:51.109624586 +0100
@@ -77,7 +77,7 @@
 }
 
 int
-main(int argc, char *argv[])
+seq_main(int argc, char *argv[])
 {
 	double start, step, end, out, dir;
 	int wflag = 0, left, right;
Index: sbase/sbase/setsid.c
===================================================================
--- sbase.orig/sbase/setsid.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/setsid.c	2022-12-06 20:18:51.112957920 +0100
@@ -13,7 +13,7 @@
 }
 
 int
-main(int argc, char *argv[])
+setsid_main(int argc, char *argv[])
 {
 	int savederrno;
 
Index: sbase/sbase/sha1sum.c
===================================================================
--- sbase.orig/sbase/sha1sum.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sha1sum.c	2022-12-06 20:18:51.116291253 +0100
@@ -21,7 +21,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sha1sum_main(int argc, char *argv[])
 {
 	int ret = 0, (*cryptfunc)(int, char **, struct crypt_ops *, uint8_t *, size_t) = cryptmain;
 	uint8_t md[SHA1_DIGEST_LENGTH];
Index: sbase/sbase/sha224sum.c
===================================================================
--- sbase.orig/sbase/sha224sum.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sha224sum.c	2022-12-06 20:18:51.119624586 +0100
@@ -21,7 +21,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sha224sum_main(int argc, char *argv[])
 {
 	int ret = 0, (*cryptfunc)(int, char **, struct crypt_ops *, uint8_t *, size_t) = cryptmain;
 	uint8_t md[SHA224_DIGEST_LENGTH];
Index: sbase/sbase/sha256sum.c
===================================================================
--- sbase.orig/sbase/sha256sum.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sha256sum.c	2022-12-06 20:18:51.122957920 +0100
@@ -21,7 +21,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sha256sum_main(int argc, char *argv[])
 {
 	int ret = 0, (*cryptfunc)(int, char **, struct crypt_ops *, uint8_t *, size_t) = cryptmain;
 	uint8_t md[SHA256_DIGEST_LENGTH];
Index: sbase/sbase/sha384sum.c
===================================================================
--- sbase.orig/sbase/sha384sum.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sha384sum.c	2022-12-06 20:18:51.122957920 +0100
@@ -21,7 +21,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sha384sum_main(int argc, char *argv[])
 {
 	int ret = 0, (*cryptfunc)(int, char **, struct crypt_ops *, uint8_t *, size_t) = cryptmain;
 	uint8_t md[SHA384_DIGEST_LENGTH];
Index: sbase/sbase/sha512-224sum.c
===================================================================
--- sbase.orig/sbase/sha512-224sum.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sha512-224sum.c	2022-12-06 20:18:51.126291252 +0100
@@ -21,7 +21,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sha512_224sum_main(int argc, char *argv[])
 {
 	int ret = 0, (*cryptfunc)(int, char **, struct crypt_ops *, uint8_t *, size_t) = cryptmain;
 	uint8_t md[SHA512_224_DIGEST_LENGTH];
Index: sbase/sbase/sha512-256sum.c
===================================================================
--- sbase.orig/sbase/sha512-256sum.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sha512-256sum.c	2022-12-06 20:18:51.129624586 +0100
@@ -21,7 +21,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sha512_256sum_main(int argc, char *argv[])
 {
 	int ret = 0, (*cryptfunc)(int, char **, struct crypt_ops *, uint8_t *, size_t) = cryptmain;
 	uint8_t md[SHA512_256_DIGEST_LENGTH];
Index: sbase/sbase/sha512sum.c
===================================================================
--- sbase.orig/sbase/sha512sum.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sha512sum.c	2022-12-06 20:18:51.132957919 +0100
@@ -21,7 +21,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sha512sum_main(int argc, char *argv[])
 {
 	int ret = 0, (*cryptfunc)(int, char **, struct crypt_ops *, uint8_t *, size_t) = cryptmain;
 	uint8_t md[SHA512_DIGEST_LENGTH];
Index: sbase/sbase/sleep.c
===================================================================
--- sbase.orig/sbase/sleep.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sleep.c	2022-12-06 20:18:51.136291252 +0100
@@ -10,7 +10,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sleep_main(int argc, char *argv[])
 {
 	unsigned seconds;
 
Index: sbase/sbase/sort.c
===================================================================
--- sbase.orig/sbase/sort.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sort.c	2022-12-06 20:18:51.139624586 +0100
@@ -323,7 +323,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sort_main(int argc, char *argv[])
 {
 	FILE *fp, *ofp = stdout;
 	struct linebuf linebuf = EMPTY_LINEBUF;
Index: sbase/sbase/split.c
===================================================================
--- sbase.orig/sbase/split.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/split.c	2022-12-06 20:18:51.139624586 +0100
@@ -45,7 +45,7 @@
 }
 
 int
-main(int argc, char *argv[])
+split_main(int argc, char *argv[])
 {
 	FILE *in = stdin, *out = NULL;
 	off_t size = 1000, n;
Index: sbase/sbase/sponge.c
===================================================================
--- sbase.orig/sbase/sponge.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sponge.c	2022-12-06 20:18:51.142957918 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sponge_main(int argc, char *argv[])
 {
 	char tmp[] = "/tmp/sponge-XXXXXX";
 	int fd, tmpfd;
Index: sbase/sbase/strings.c
===================================================================
--- sbase.orig/sbase/strings.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/strings.c	2022-12-06 20:18:51.146291252 +0100
@@ -49,7 +49,7 @@
 }
 
 int
-main(int argc, char *argv[])
+strings_main(int argc, char *argv[])
 {
 	FILE *fp;
 	size_t min = 4;
Index: sbase/sbase/sync.c
===================================================================
--- sbase.orig/sbase/sync.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/sync.c	2022-12-06 20:18:51.149624585 +0100
@@ -10,7 +10,7 @@
 }
 
 int
-main(int argc, char *argv[])
+sync_main(int argc, char *argv[])
 {
 	ARGBEGIN {
 	default:
Index: sbase/sbase/tail.c
===================================================================
--- sbase.orig/sbase/tail.c	2022-12-06 19:10:45.126426470 +0100
+++ sbase/sbase/tail.c	2022-12-06 20:18:51.152957918 +0100
@@ -147,7 +147,7 @@
 }
 
 int
-main(int argc, char *argv[])
+tail_main(int argc, char *argv[])
 {
 	struct stat st1, st2;
 	int fd;
Index: sbase/sbase/tar.c
===================================================================
--- sbase.orig/sbase/tar.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/tar.c	2022-12-06 20:18:51.152957918 +0100
@@ -512,7 +512,7 @@
 }
 
 int
-main(int argc, char *argv[])
+tar_main(int argc, char *argv[])
 {
 	struct recursor r = { .fn = c, .follow = 'P', .flags = DIRFIRST };
 	struct stat st;
Index: sbase/sbase/tee.c
===================================================================
--- sbase.orig/sbase/tee.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/tee.c	2022-12-06 20:18:51.156291251 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+tee_main(int argc, char *argv[])
 {
 	int *fds = NULL;
 	size_t i, nfds;
Index: sbase/sbase/test.c
===================================================================
--- sbase.orig/sbase/test.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/test.c	2022-12-06 20:18:51.159624585 +0100
@@ -229,7 +229,7 @@
 }
 
 int
-main(int argc, char *argv[])
+test_main(int argc, char *argv[])
 {
 	int (*narg[])(char *[]) = { noarg, onearg, twoarg, threearg, fourarg };
 	size_t len;
Index: sbase/sbase/tftp.c
===================================================================
--- sbase.orig/sbase/tftp.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/tftp.c	2022-12-06 20:18:51.162957918 +0100
@@ -247,7 +247,7 @@
 }
 
 int
-main(int argc, char *argv[])
+tftp_main(int argc, char *argv[])
 {
 	struct addrinfo hints, *res, *r;
 	struct timeval tv;
Index: sbase/sbase/time.c
===================================================================
--- sbase.orig/sbase/time.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/time.c	2022-12-06 20:18:51.166291251 +0100
@@ -15,7 +15,7 @@
 }
 
 int
-main(int argc, char *argv[])
+time_main(int argc, char *argv[])
 {
 	pid_t pid;
 	struct tms tms; /* user and sys times */
Index: sbase/sbase/touch.c
===================================================================
--- sbase.orig/sbase/touch.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/touch.c	2022-12-06 20:18:51.166291251 +0100
@@ -106,7 +106,7 @@
 }
 
 int
-main(int argc, char *argv[])
+touch_main(int argc, char *argv[])
 {
 	struct stat st;
 	char *ref = NULL;
Index: sbase/sbase/tr.c
===================================================================
--- sbase.orig/sbase/tr.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/tr.c	2022-12-06 20:18:51.169624584 +0100
@@ -170,7 +170,7 @@
 }
 
 int
-main(int argc, char *argv[])
+tr_main(int argc, char *argv[])
 {
 	Rune r, lastrune = 0;
 	size_t off1, off2, i, m;
Index: sbase/sbase/true.c
===================================================================
--- sbase.orig/sbase/true.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/true.c	2022-12-06 20:18:51.172957918 +0100
@@ -1,6 +1,6 @@
 /* See LICENSE file for copyright and license details. */
 int
-main(void)
+true_main(void)
 {
 	return 0;
 }
Index: sbase/sbase/tsort.c
===================================================================
--- sbase.orig/sbase/tsort.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/tsort.c	2022-12-06 20:18:51.176291251 +0100
@@ -179,7 +179,7 @@
 }
 
 int
-main(int argc, char *argv[])
+tsort_main(int argc, char *argv[])
 {
 	FILE *fp = stdin;
 	const char *fn = "<stdin>";
Index: sbase/sbase/tty.c
===================================================================
--- sbase.orig/sbase/tty.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/tty.c	2022-12-06 20:18:51.179624584 +0100
@@ -11,7 +11,7 @@
 }
 
 int
-main(int argc, char *argv[])
+tty_main(int argc, char *argv[])
 {
 	char *tty;
 
Index: sbase/sbase/uname.c
===================================================================
--- sbase.orig/sbase/uname.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/uname.c	2022-12-06 20:18:51.179624584 +0100
@@ -12,7 +12,7 @@
 }
 
 int
-main(int argc, char *argv[])
+uname_main(int argc, char *argv[])
 {
 	struct utsname u;
 	int mflag = 0, nflag = 0, rflag = 0, sflag = 0, vflag = 0;
Index: sbase/sbase/unexpand.c
===================================================================
--- sbase.orig/sbase/unexpand.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/unexpand.c	2022-12-06 20:18:51.182957917 +0100
@@ -129,7 +129,7 @@
 }
 
 int
-main(int argc, char *argv[])
+unexpand_main(int argc, char *argv[])
 {
 	FILE *fp;
 	int ret = 0;
Index: sbase/sbase/uniq.c
===================================================================
--- sbase.orig/sbase/uniq.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/uniq.c	2022-12-06 20:18:51.186291251 +0100
@@ -98,7 +98,7 @@
 }
 
 int
-main(int argc, char *argv[])
+uniq_main(int argc, char *argv[])
 {
 	FILE *fp[2] = { stdin, stdout };
 	int ret = 0, i;
Index: sbase/sbase/unlink.c
===================================================================
--- sbase.orig/sbase/unlink.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/unlink.c	2022-12-06 20:18:51.189624584 +0100
@@ -10,7 +10,7 @@
 }
 
 int
-main(int argc, char *argv[])
+unlink_main(int argc, char *argv[])
 {
 	ARGBEGIN {
 	default:
Index: sbase/sbase/uudecode.c
===================================================================
--- sbase.orig/sbase/uudecode.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/uudecode.c	2022-12-06 20:18:51.192957917 +0100
@@ -224,7 +224,7 @@
 }
 
 int
-main(int argc, char *argv[])
+uudecode_main(int argc, char *argv[])
 {
 	FILE *fp = NULL, *nfp = NULL;
 	mode_t mode = 0;
Index: sbase/sbase/uuencode.c
===================================================================
--- sbase.orig/sbase/uuencode.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/uuencode.c	2022-12-06 20:18:51.192957917 +0100
@@ -97,7 +97,7 @@
 }
 
 int
-main(int argc, char *argv[])
+uuencode_main(int argc, char *argv[])
 {
 	FILE *fp = NULL;
 	void (*uuencode_f)(FILE *, const char *, const char *) = uuencode;
Index: sbase/sbase/wc.c
===================================================================
--- sbase.orig/sbase/wc.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/wc.c	2022-12-06 20:18:51.196291250 +0100
@@ -67,7 +67,7 @@
 }
 
 int
-main(int argc, char *argv[])
+wc_main(int argc, char *argv[])
 {
 	FILE *fp;
 	int many;
Index: sbase/sbase/which.c
===================================================================
--- sbase.orig/sbase/which.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/which.c	2022-12-06 20:18:51.199624583 +0100
@@ -69,7 +69,7 @@
 }
 
 int
-main(int argc, char *argv[])
+which_main(int argc, char *argv[])
 {
 	char *path;
 	int found = 0, foundall = 1;
Index: sbase/sbase/whoami.c
===================================================================
--- sbase.orig/sbase/whoami.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/whoami.c	2022-12-06 20:18:51.202957917 +0100
@@ -13,7 +13,7 @@
 }
 
 int
-main(int argc, char *argv[])
+whoami_main(int argc, char *argv[])
 {
 	uid_t uid;
 	struct passwd *pw;
Index: sbase/sbase/xargs.c
===================================================================
--- sbase.orig/sbase/xargs.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/xargs.c	2022-12-06 20:18:51.202957917 +0100
@@ -193,7 +193,7 @@
 }
 
 int
-main(int argc, char *argv[])
+xargs_main(int argc, char *argv[])
 {
 	int ret = 0, leftover = 0, i;
 	size_t argsz, argmaxsz;
Index: sbase/sbase/xinstall.c
===================================================================
--- sbase.orig/sbase/xinstall.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/xinstall.c	2022-12-06 20:18:51.206291250 +0100
@@ -81,7 +81,7 @@
 }
 
 int
-main(int argc, char *argv[])
+xinstall_main(int argc, char *argv[])
 {
 	int dflag = 0;
 	char *gflag = 0;
Index: sbase/sbase/yes.c
===================================================================
--- sbase.orig/sbase/yes.c	2022-12-06 19:10:45.129759803 +0100
+++ sbase/sbase/yes.c	2022-12-06 20:18:51.209624583 +0100
@@ -10,7 +10,7 @@
 }
 
 int
-main(int argc, char *argv[])
+yes_main(int argc, char *argv[])
 {
 	const char *s;
 
